### ISSUE TITLE
Vizier query results differ from the results of astroquery

### ISSUE THREAD

**Issue by boada** [astropy/astroquery#915]
I noticed this strange behavior while digging deep into the [photometrypipeline](https://github.com/mommermi/photometrypipeline) project. After discussing with @mommermi, he came up with the example below. 

I want to get some data from SDSS dr9. Specifically, I want i-band photometry on [this object](http://skyserver.sdss.org/dr13/en/tools/explore/summary.aspx?id=1237679997155541093). That's a dr13 link because the images don't always show up on dr9 links. I can use the following code (from @mommermi)

    from astroquery.vizier import Vizier
    import astropy.coordinates as coord
    import astropy.units as u

    vquery = Vizier()

    field = coord.SkyCoord(ra=329.50922672, dec=-2.33703204, unit=(u.deg, u.deg),
                       frame='icrs')

    data = vquery.query_region(field,
                           width=("%fd" % 0.001),
                           catalog="V/139/sdss9")[0]
                                                 
    print(data.columns)
    print(data['SDSS9', 'RAJ2000', 'DEJ2000', 'gmag', 'rmag', 'imag', 'zmag', 'Q'])

The result is two objects *with the same identifier*. One is a star (the object I want) and one is a nearby galaxy. Both objects are given as having good photometry (Q=3). However, the SDSS dr9 webpage says the star's photometry is good and the galaxy is bad. So which is the object that I want!?

>      SDSS9                          RAJ2000    DEJ2000    gmag   rmag   imag   zmag   Q 
>                                        deg        deg      mag    mag    mag    mag      
>     J215802.21-022013.3 329.509226  -2.337048 19.549 18.874 18.604 18.392   3
>     J215802.21-022013.3 329.509227  -2.337032 19.635 18.971 18.675 18.561   3

Interestingly, when you query vizier directly, you get different results. Using [this query](http://vizier.u-strasbg.fr/viz-bin/VizieR?-source=V/139&-c=329.50922672,-2.33703204&-c.rs=10&-to=3) (you have to click submit), you also get two objects. However this time each has a unique identifier. One of which is correctly labeled as a star(cl=6) (that's the object I want) and one is the pesky nearby galaxy (cl=3). 

Troublingly but probably unrelated, vizier says the photometry of the galaxy is good (Q=3) when the warning flags from sloan say otherwise.

TL;DR: astroquery gives different results when compared to querying vizier directly.

**Comment by keflavich** [comment]
You've made a small error in your query that *partly* explains the results.  You used `width` instead of `radius`; when you did the webform query you used radius=10 arcsec.  The equivalent from astroquery:
```
In [17]: data = vquery.query_region(field,
                       radius=10*u.arcsec,
                       catalog="V/139/sdss9")[0]

In [18]: data
Out[18]:
<Table masked=True length=3>
 mode q_mode   cl         SDSS9        m_SDSS9   Im    RAJ2000    DEJ2000    ObsDate    Q   ...   gmag   e_gmag   rmag   e_rmag   imag   e_imag   zmag   e_zmag   zsp
                                                         deg        deg         yr          ...   mag     mag     mag     mag     mag     mag     mag     mag
uint8 bytes1 uint8       bytes19        bytes1 bytes2  float64    float64    float64  uint8 ... float32 float32 float32 float32 float32 float32 float32 float32 float64
----- ------ ----- ------------------- ------- ------ ---------- ---------- --------- ----- ... ------- ------- ------- ------- ------- ------- ------- ------- -------
    1            3 J215802.13-022016.2             Im 329.508894  -2.337837 2009.7092     3 ...  22.333   0.194  20.826   0.079  20.734   0.110  19.980   0.235      --
    2            6 J215802.21-022013.3       *     Im 329.509226  -2.337048 2008.8819     3 ...  19.549   0.016  18.874   0.012  18.604   0.012  18.392   0.038      --
    1      +     6 J215802.21-022013.3             Im 329.509227  -2.337032 2009.7092     3 ...  19.635   0.014  18.971   0.011  18.675   0.012  18.561   0.034      --
```

I'm not sure why there are three entries from astroquery and only two from the webform, but at least both of your targets show up now.

**Comment by boada** [comment]
Thanks for point that out. I still find it strange that the star is listed twice (both with good photometry) at the same position, but the photometry values differ by ~0.1 mags (for the r-band).

**Comment by boada** [comment]
Ok, I bet I get it now. 

The star and the galaxy are blended. So that's why there are three entries. From the top down, the deblended galaxy, the star-galaxy blend, and then the deblended star. That's why the only object with actual "clean" photometry (denoted by q_mode = +) is the third entry. The deblend on the galaxy wasn't good enough. So while the imaging is ok (Q=3) the photometry isn't good because it is missing the +.

**Comment by mommermi** [comment]
Ok, that makes sense. But it still seems weird to me that both stars share the same `SDSS9` id. We can probably avoid this ambiguity by only keeping good photometry sources (denoted by `+`). Thanks , guys, for looking into this!

**Comment by boada** [comment]
They have the same SDSS name, but if you look they have three different objID's. 

> In [5]: data['objID']
> Out[5]: 
> <MaskedColumn name='objID' dtype='int64' description='SDSS unique object identifier (2) >
> {\\fg{DarkGreen}(links to SDSS-DR9 details)}' length=3>
> 1237679997155541094
> 1237679026501189691
> 1237679997155541093

**Comment by keflavich** [comment]
To reproduce the astroquery result, set `mode=`[blank] in the Vizier web query form; the default is `=1` which selects the 'primary' object, which I assume is the one with good photometry.

To get exactly the result you expected from the online query form, you can do:
```
In [8]: data = vquery(column_filters={'mode':'=1'}).query_region(field,
                       radius=10*u.arcsec,
                       catalog="V/139/sdss9")[0]

In [9]: data
Out[9]:
<Table masked=True length=2>
 mode q_mode   cl         SDSS9        m_SDSS9   Im    RAJ2000    DEJ2000    ObsDate    Q   ...   gmag   e_gmag   rmag   e_rmag   imag   e_imag   zmag   e_zmag   zsp
                                                         deg        deg         yr          ...   mag     mag     mag     mag     mag     mag     mag     mag
uint8 bytes1 uint8       bytes19        bytes1 bytes2  float64    float64    float64  uint8 ... float32 float32 float32 float32 float32 float32 float32 float32 float64
----- ------ ----- ------------------- ------- ------ ---------- ---------- --------- ----- ... ------- ------- ------- ------- ------- ------- ------- ------- -------
    1            3 J215802.13-022016.2             Im 329.508894  -2.337837 2009.7092     3 ...  22.333   0.194  20.826   0.079  20.734   0.110  19.980   0.235      --
    1      +     6 J215802.21-022013.3             Im 329.509227  -2.337032 2009.7092     3 ...  19.635   0.014  18.971   0.011  18.675   0.012  18.561   0.034      --
```
